### PR TITLE
Store the last state of window geometry of an electron app.

### LINKED ISSUE #3535: [electron] Store and reuse the window size between sessions
The electron application should store and reuse the window size between sessions. It should be stored per workspace.

### PR DESCRIPTION
**Pull request by jbicker**
Regarding #3535 

For now the last state of a resized, moved or closed window gets stored and reused for the next opening window. It is not per workspace, yet, because the URL we get [here](https://github.com/theia-ide/theia/blob/jb/GH-3535/dev-packages/application-manager/src/generator/frontend-generator.ts#L144) is undefined if we open a fresh electron app. But it gets open with the last workspace. I don't know where that comes from. Can anyone explain?
OTOH it feels good enough to just restore the last position and size independent from WS.

### CODE REVIEW

File: examples/electron/package.json [hunk 1]
```diff
@@ -55,7 +55,8 @@
     "@theia/typescript": "^0.3.19",
     "@theia/userstorage": "^0.3.19",
     "@theia/variable-resolver": "^0.3.19",
-    "@theia/workspace": "^0.3.19"
+    "@theia/workspace": "^0.3.19",
+    "electron-store": "^2.0.0"
```
Comment (kittaakos) [inline examples/electron/package.json]: I think if you put the dependency here then it won't work for other extensions.

Also, don't we need a CQ for the new dependency?

File: dev-packages/application-manager/src/generator/frontend-generator.ts [hunk 2]
```diff
@@ -161,6 +166,19 @@ if (isMaster) {
                 shell.openExternal(url);
             });
 
+            const safeState = ()=>{
```
Comment (kittaakos) [inline dev-packages/application-manager/src/generator/frontend-generator.ts]: `const safeState = () => {`

**Comment by jbicker** [comment]
> (I think it is not fair to include `Fixes #3535 (partly).` because merging the PR closes the issue although it is not fixed.)

Sorry, you're right. Was not meant like that.

File: examples/electron/package.json (hunk 1 above)
Comment (jbicker) [reply examples/electron/package.json]: Ok, moved that to core. 
I don't know about CQ. What is to do there?

Comment (kittaakos) [reply examples/electron/package.json]: https://github.com/theia-ide/theia/wiki/Registering-CQs

Comment (kittaakos) [reply examples/electron/package.json]: But let's ask @marcdumais-work whether we need a new CQ for this or not. Thanks!

**Comment by jbicker** [comment]
Regarding CQ: https://spectrum.chat/theia/dev/cq-for-new-dependency~2d11bf0d-65f8-4b3d-850c-c7d8758b44ed

**Pull request merged**